server/applicationServer.go

```diff
@@ -0,0 +1,136 @@
+package server
+
+import (
+	"context"
+	"log"
+	"net/http"
+	"os"
+	"time"
+)
+
+// NewServer creates an instance of the app server
+func NewServer() *ApplicationServer {
+	server := &ApplicationServer{
+		Config: Config,
+		Logger: log.New(os.Stdout, "", log.Ldate|log.Ltime),
+		router: http.NewServeMux(),
+	}
+
+	server.httpServer = http.Server{
+		Addr:    server.Config.ServerAddress,
+		Handler: server.router,
+	}
+
+	return server
+}
+
+// Start will start the server eventually
+func (s *ApplicationServer) Start(addr string) error {
+	s.Config.ServerAddress = addr
+	s.httpServer.Addr = s.Config.ServerAddress
+
+	err := s.httpServer.ListenAndServe()
+	return err
+}
+
+// Stop will stop the server eventually
+func (s *ApplicationServer) Stop() error {
+	err := s.httpServer.Shutdown(context.Background())
+	return err
+}
+
+// Get adds a handler for the 'GET' http method for server s.
+func (s *ApplicationServer) Get(route string, f func(http.ResponseWriter, *http.Request)) {
+	s.router.HandleFunc(route, s.get(f))
+}
+
+func (s *ApplicationServer) get(h http.HandlerFunc) http.HandlerFunc {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method 'get' differs only by capitalization to method 'Get' in the same source file](https://app.codacy.com/gh/hrple/common/pullRequest?prid=5175046)